src/nspawn/nspawn.c

```diff
@@ -1395,6 +1395,12 @@ static int copy_devnodes(const char *dest) {
 
                 } else {
                         if (mknod(to, st.st_mode, st.st_rdev) < 0) {
+                                /*
+                                 * This is some sort of protection too against
+                                 * recursive userns chown on shared /dev/
```

Comment:
Hm, I don't understand this comment. What does recursion does "recursive" refer to?

Also, if errno == EPERM, than the new log_notice could be emitted unnecessarily. You should move the log_notice inside of `if (errno != EPERM)` block. Also, it'd be nicer to emit the notice *after* the error, we usually follow this order of error first, hint second.